ouroboros-network/test/Test/Ouroboros/Network/Testnet.hs

```diff
@@ -137,6 +137,8 @@ tests =
   , testProperty "cm & ig timeouts enforced"
                  prop_diffusion_timeouts_enforced
   , testProperty "unit #4177" unit_4177
+  , testProperty "never connect to self"
```

Comment:
What is so about self-connecting? (see comment earlier)

Reply:
This is more a test for the test suite as our network simulation isn't really capable of dealing with self connects, so this test is to catch such cases and not have them appear in other tests disguised